.github/workflows/e2e.yml

```diff
@@ -0,0 +1,42 @@
+name: E2E Tests
+
+on:
+  pull_request:
+    branches: [ master, release/** ]
+  push:
+    branches: [ master, release/** ]
+    tags: [ v* ]
+
+jobs:
+  Run-E2E-Tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        packages: ['contract.opcodes,contract.openzeppelin',
+                   'contract.precompile,contract.records',
+                   'contract.hapi']
+    steps:
+      - uses: actions/checkout@v2 # checkout current branch
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+      - name: Build docker images # build the image for hedera-node
+        run: bash compose-build.sh
+      - name: Start network # start 3 node network
+        run: docker compose up -d
+      - name: Healthcheck for Hedera node # wait until node is ready to serve requests
+        working-directory: github-actions-utils
+        run: bash healthcheck.sh
+      - name: Run E2E Suites # run selected e2e suites
+        working-directory: github-actions-utils
+        run: bash run-e2e-tests.sh ${{matrix.packages}}
+      - name: Upload surefire report
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: surefire-report-(${{matrix.packages}})
+          path: test-clients/target/surefire-reports/*
```

Comment:
Consider adding a test result job using [publish-unit-test-result-action](https://github.com/EnricoMi/publish-unit-test-result-action#use-with-matrix-strategy) at the end that aggregates the results into a PR comment:

```yaml
  test-results:
    name: "Tests Results"
    needs: Run-E2E-Tests
    runs-on: ubuntu-latest
    if: always()

    steps:
      - name: Download Artifacts
        uses: actions/download-artifact@v2
        with:
          path: artifacts

      - name: Publish Results
        uses: EnricoMi/publish-unit-test-result-action@v1
        with:
          files: artifacts/**/*.xml
```